File: ios/App/App/SwipePlugin.m
//
//  SwipePlugin.m
//  App
//
//  Created by Aleksandr Stekolshchikov on 17.03.2022.
//

#import <Capacitor/Capacitor.h>

CAP_PLUGIN(SwipeBackPlugin, "SwipeBack",
    CAP_PLUGIN_METHOD(enable, CAPPluginReturnPromise);
    CAP_PLUGIN_METHOD(disable, CAPPluginReturnPromise);
)

File: ios/App/App/SwipePlugin.swift
import Capacitor

@objc(SwipeBackPlugin)
public class SwipeBackPlugin: CAPPlugin {
    @objc func enable(_ call: CAPPluginCall) {
        self.bridge?.webView?.allowsBackForwardNavigationGestures = true
        call.resolve([ "error": 0 ]);
    }
        
    @objc func disable(_ call: CAPPluginCall) {
        self.bridge?.webView?.allowsBackForwardNavigationGestures = false
        call.resolve([ "error": 0 ]);
    }
}
